src/users/users.service.ts

```diff
@@ -70,4 +74,11 @@ export class UsersService {
       },
     };
   }
+
+  async findAll() {
+    const users = await this.repository.findAll();
+    return await Promise.all(
+      users.map((user) => this.generateUserResponse(user)),
+    );
+  }
```

Comment:
I see two problems with this:
 - It'll generate a valid token for ALL users, which is probably not a good thing to do regardless if it's exposed in an API or not
 - Although the spec doesn't talk about a functionality like this, but I think the returned data format should be changed. Instead of a list of singular user objects (`{user: {...}}[]`) it could be one plural users object, e.g. `{users: [user1, user2]}`